subcommand/controller/command.go

```diff
@@ -0,0 +1,117 @@
+package controller
+
+import (
+	"context"
+	"flag"
+	"os"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/hashicorp/consul-ecs/awsutil"
+	"github.com/hashicorp/consul-ecs/controller"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/go-hclog"
+	"github.com/mitchellh/cli"
+)
+
+const (
+	flagAgentSecretARN   = "agent-secret-arn"
+	flagSecretNamePrefix = "secret-name-prefix"
+
+	consulCACertEnvVar = "CONSUL_CACERT_PEM"
+)
+
+type Command struct {
+	UI                   cli.Ui
+	flagAgentSecretARN   string
+	flagSecretNamePrefix string
+
+	log     hclog.Logger
+	flagSet *flag.FlagSet
+	once    sync.Once
+}
+
+func (c *Command) init() {
+	c.flagSet = flag.NewFlagSet("", flag.ContinueOnError)
+	c.flagSet.StringVar(&c.flagAgentSecretARN, flagAgentSecretARN, "", "ARN of AWS Secrets Manager secret")
+	c.flagSet.StringVar(&c.flagSecretNamePrefix, flagSecretNamePrefix, "", "The prefix for secret names stored in AWS Secrets Manager")
+
+	c.log = hclog.New(nil)
+}
+
+func (c *Command) Run(args []string) int {
+	c.once.Do(c.init)
+	if err := c.flagSet.Parse(args); err != nil {
+		return 1
+	}
+
+	err := c.run()
+	if err != nil {
+		c.log.Error(err.Error())
+		return 1
+	}
+	return 0
+}
+
+func (c *Command) run() error {
```

Comment:
I think the pattern I've seen is using `realRun` or something like that. (I didn't ever know Go's functions were case sensitive!)

Reply:
Oh sorry, I didn't realize it was a pattern! I changed it because personally I found it a bit confusing because it made me think that the `Run()` method is not "real" 😸.  Where does the `realRun` pattern come from? I don't think I've seen it before. 

Yeah, golang is case-sensitive. That's how it decides if a function is exported or not. I've seen `run()` method in the Consul codebase being used a few times, e.g. [here](https://github.com/hashicorp/consul/blob/a2e6a15e936b4907bf6a1790d91da96ee8949997/command/connect/envoy/envoy.go#L229), but don't think I can call it a pattern.

Reply:
I was taking it from TF (https://github.com/hashicorp/terraform/blob/main/main.go#L61). But if Consul already has their own pattern then it makes more sense to use theirs.